monkey/monkey_island/cc/dal/IEdgeRepository.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC
+from typing import Optional, Sequence
+
+from monkey_island.cc.models.edge import Edge
+
+
+class IEdgeRepository(ABC):
```

Comment:
I have a feeling we may want to group edges and nodes together in `INetworkRepository` or `IGraphRepository` or some such thing.

Reply:
Machines and edges will likely remain separate entities. Not sure what grouping them would achieve apart from obfuscating the entity names

Reply:
Edges aren't useful on their own. An edge means nothing without the nodes it connects. 

Machines are useful on their own, having the machine repository probably makes sense.

Rather than providing an interface to get edges in all sorts of ways (by src, by dest, etc.), why don't we provide an interface to submit edges and get back a map? Then the map object can provide ways to get data about edges and nodes. If the `IMapRepository` uses `IMachineRepository` under the hood, that's fine.

Let's discuss it and brainstorm a little bit. I think providing a richer `NetworkMap` object will take some of the burden off of the calling code and make the repository interface cleaner.